sdk/appcenter-analytics/src/main/java/com/microsoft/appcenter/analytics/channel/SessionTracker.java

```diff
@@ -127,14 +86,14 @@ public void onEnqueuingLog(@NonNull Log log, @NonNull String groupName) {
          */
         Date timestamp = log.getTimestamp();
         if (timestamp != null) {
-            Map.Entry<Long, UUID> pastSession = mSessions.lowerEntry(timestamp.getTime());
+            SessionContext.SessionInfo pastSession = SessionContext.getInstance().getSessionAt(timestamp.getTime());
             if (pastSession != null) {
-                log.setSid(pastSession.getValue());
+                log.setSid(pastSession.getSessionId());
             }
         }
 
-        /* If the log is not correlated to a past session. */
-        if (log.getSid() == null) {
+        /* If the log is not to be correlated to a past session, assign one. */
```

Comment:
Assign one what? As far as I can see, the log doesn't get changed at all. I think I'm misunderstanding the comment.

Reply:
I can modify the comment